app.js

```diff
@@ -44,25 +47,74 @@ app.Views.TodoAppView = Backbone.View.extend({
         this.todoContent.append(this.subviews.todoListView.render());
 
         return this.$el;
+    },
+    createTodoOnEnter: function(e) {
+        const title = this.newTodoField.val().trim();
+        if (e.which == 13 && title != "") {
+
+            this.subviews.todoListView.collection.add(new app.Models.TodoItem({
+                title: title
+            }));
+        }
     }
 });
 app.Views.ToDoItemView = Backbone.View.extend({
     tagName: "li",
     className: "todo-item",
+    initialize: function() {
+        // this.listenTo(this.model, "destroy", this.remove);  // is this needed even though list view already listens for remove
+        this.listenTo(this.model, "change", this.render);
+    },
+    events: {
+        "click .js-delete-btn": "delete",
+        "dblclick .js-item-label": "startEditing",
+        "blur .js-edit-input": "finishEditing",        
+        "keypress .js-edit-input": "finishEditingOnEnter",
+        "change .js-complete-checkbox": "updateCompleted"
+    },
     template: Handlebars.compile($("#todo-item-template").html()),
     render: function() {
         console.log("rendering item view: ", this.model);
 
         this.$el.html(this.template(this.model.toJSON()))
+
+        this.editInput = this.$(".js-edit-input");
+        this.completeCheckbox = this.$(".js-complete-checkbox");
+
         return this.$el;
+    },
+    delete: function() {
+        this.model.destroy();
+    },
+    startEditing: function() {
+        this.editInput.val(this.model.get("title"));
+        this.$el.addClass("editing");
+        this.editInput.focus();
+    },
+    finishEditingOnEnter: function(e) {
+        if (e.which == 13) {
+            this.finishEditing();
+        }
+    },
+    finishEditing: function() {
+        this.model.set({
+            title: this.editInput.val().trim()
+        })
+
+        this.$el.removeClass("editing");
+    },
+    updateCompleted: function() {
+        this.model.set({
+            completed: this.completeCheckbox.prop("checked")
+        });
     }
 });
 app.Views.ToDoListView = Backbone.View.extend({
     tagName: "ul",
     className: ".todo-items",
     initialize: function() {
-        // this.listenTo(this.collection, "add", );
-        // this.listenTo(this.collection, "remove", );
+        this.listenTo(this.collection, "add", this.render);
```

Comment:
You can combine these two listen to's into a single one instead since they are both listening to the same collection and calling the same event